version: "3.7"

services:
  app:
    build:
      context: .
      dockerfile: ./deployments/production/go/Dockerfile
    image: go_fast_production_app
    restart: on-failure
    command:     command:
      [
        "/root/wait-for",
        "postgres:5432",
        "/root/webserver",
        "-p",
        "10000",
      ]
    depends_on:
      - postgres
    env_file:
      - ./.envs/.production/.postgres
    expose:
      - "10000"
    networks:
      - go_fast_production
      - default
    labels:
      - "traefik.docker.network=go_fast_production"
      - "traefik.enable=true"
      - "traefik.basic.frontend.rule=Host:go-fast.djangulo.com"
      - "traefik.basic.port=10000"
      - "traefik.basic.protocol=http"
      - "traefik.admin.frontend.rule=Host:go-fast-admin.djangulo.com"
      - "traefik.admin.protocol=https"
      - "traefik.admin.port=10443"
      - "traefik.admin.frontend.auth.basic.users=djangulo:$$apr1$$h.wPwHEW$$yEgYhPyNr4sn8dy4Nbbs70"

  postgres:
    build:
      context: .
      dockerfile: ./deployments/production/postgres/Dockerfile
    healthcheck:
      test: ["CMD-SHELL", "pg_isready"]
      interval: 1m30s
      timeout: 10s
      retries: 5
      start_period: 30s
    restart: always
    image: go_fast_production_postgres
    volumes:
      - production_postgres_data:/var/lib/postgresql/data
      - production_postgres_backups:/backups
    env_file:
      - ./.envs/.production/.postgres

volumes:
  production_postgres_data: {}
  production_postgres_backups: {}

networks:
  go_fast_production:
    external: true
